xdsl/dialects/arith.py

```diff
@@ -113,7 +113,6 @@ def from_float_and_width(val: float | FloatAttr[_FloatTypeT],
         return Constant.create(result_types=[typ], attributes={"value": val})
 
 
-@dataclass
```

Comment:
Was that unrelated?

I guess if it's unrelated, we could still keep that!

Reply:
Totally unrelated! I was playing around trying to fetch everything in BinaryOperation (turns out it's just a slight pain with the different constraints, which are variables or types :/ I still think there's a quest to have there too!)
Forgot about it as it rang no alarm at all. No feelings about it, but confused about ambigous negation here : are you saying I should keep the decorator, or keep its deletion ? :yum: